v3/lints/cabf_br/lint_subject_common_name_not_exactly_from_san_test.go

```diff
@@ -0,0 +1,61 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/test"
+)
+
+func TestCnNotExactlyFromSAN(t *testing.T) {
+	var testCases = []struct {
+		name      string
+		inputFile string
+
+		expectedOutput lint.LintStatus
+	}{
+		{
+			name:           "Pass - commonName in SAN.DNSNames",
+			inputFile:      "SANWithCNSeptember2021.pem",
+			expectedOutput: lint.Pass,
+		},
+		{
+			name:           "Error - common name not in SAN.DNSNames",
+			inputFile:      "SANWithoutCNSeptember2021.pem",
+			expectedOutput: lint.Error,
+		},
+		{
+			name:           "Error - common name in SAN.DNSNames but case mismatch",
+			inputFile:      "SANCaseNotMatchingCNSeptember2021.pem",
+			expectedOutput: lint.Error,
+		},
```

Comment:
Some IPv4 / IPv6 tests would be good, right?

Potentially testing some of the non-standard Ipv4 encodings (e.g. `"0x7F000001"` for a commonName, or `"[::00:01]"`)

How does Golang's stringifier handle this? Does it bracket encode as in URL encoding, or does it omit the brackets? I'm guessing it omits them. Here, again, the BRs and relevant specs are (I believe) a bit silent on what to expect, and I legitimately have no clue what systems do.

To be clear: I'm think I'm fine taking a conservative/common-sense approach here, and letting CAs raise concerns if it ends up overly restrictive (and clarifying the BRs as appropriate). I filed https://github.com/cabforum/servercert/issues/301 to track this

Reply:
Ah, @CBonnell helpfully pointed out on the issue that I'm an idiot who can't read (granted, he didn't say it, but it's an accurate self-assessment :P )

So I think the actionable thing here is still just to make sure we have tests covering the cases mentioned in https://github.com/cabforum/servercert/pull/285/files#diff-e0ac1bd190515a4f2ec09139d395ef6a8c7e9e5b612957c1f5a2dea80c6a6cfeR2092

Reply:
I believe I've now covered the test cases for correctly formatted CN/SAN IP addresses on IPv4 and v6.